### PR TITLE
Refactor chunkserverentry use std list and smart pointers

### PR DESCRIPTION
**Pull request by lgsilva3087**
Partial refactor of ChunkserverEntry related structs.
Main goals:
- Avoid C-like linked lists and use std::list.
- Simplified memory management (malloc and free was used before).
- Reuse of already allocated memory.
- Use of move semantics.
- Simplified destruction and cleanup.
- Shorter code.
- Remove void* and casts.

### CODE REVIEW

File: src/chunkserver/network_worker_thread.cc [hunk 1]
```diff
@@ -1633,35 +1585,17 @@ void NetworkWorkerThread::operator()() {
 void NetworkWorkerThread::terminate() {
 	TRACETHIS();
 	job_pool_delete(bgJobPool_);
+
```
Comment (rolysr) [inline src/chunkserver/network_worker_thread.cc:1588]: Should be considered to keep these extra spaces in the changes?

**Review by rolysr — APPROVED** [review]
LGTM. Just one suggestion to consider.

File: src/chunkserver/network_worker_thread.cc (hunk 1 above)
Comment (lgsilva3087) [reply src/chunkserver/network_worker_thread.cc:1588]: Hi Roly, if you mean the empty lines, they are intentional, to visually separate the code blocks. IMO it is easier to read this way.

**Review by ralcolea — DISMISSED** [review]
Great job @lgsilva3087 :+1: :fire: :rocket:

**Review by ralcolea — APPROVED** [review]
LGTM :+1: :fire: :rocket:

**Review by rolysr — APPROVED** [review]

**Pull request merged**